lms/djangoapps/teams/views.py

```diff
@@ -821,6 +828,87 @@ def delete(self, request, team_id):
         return Response(status=status.HTTP_204_NO_CONTENT)
 
 
+class TeamsAssignmentsView(GenericAPIView):
+    """
+        **Use Cases**
+
+            Get a team's assignments
+
+        **Example Requests**:
+
+            GET /api/team/v0/teams/{team_id}/assignments
+
+        **Response Values for GET**
+
+            If the user is logged in, the response is an array of the following data strcuture:
+
+                * display_name: The name of the assignment to display (currently the Unit title)
+
+                * location: The jump link to a specific assignments
+
+            For all text fields, clients rendering the values should take care
+            to HTML escape them to avoid script injections, as the data is
+            stored exactly as specified. The intention is that plain text is
+            supported, not HTML.
+
+            If team assignments are not enabled for course, a 503 is returned.
+
+            If the user is not logged in, a 401 error is returned.
+
+            If the user is unenrolled or does not have API access, a 403 error is returned.
+
+            If the supplied course/team is bad or the user is not permitted to
+            search in a protected team, a 404 error is returned as if the team does not exist.
+
+    """
+    authentication_classes = (BearerAuthentication, SessionAuthentication)
+    permission_classes = (
+        permissions.IsAuthenticated,
+        IsEnrolledOrIsStaff,
+        HasSpecificTeamAccess,
+        IsStaffOrPrivilegedOrReadOnly,
+    )
+
+    def get(self, request, team_id):
+        """GET v0/teams/{team_id_pattern}/assignments"""
+        course_team = get_object_or_404(CourseTeam, team_id=team_id)
+        user = request.user
+        course_id = course_team.course_id
+
+        if not are_team_submissions_enabled(course_id):
+            return Response(status=status.HTTP_503_SERVICE_UNAVAILABLE)
+
+        if not has_team_api_access(request.user, course_id):
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        if not has_specific_team_access(user, course_team):
```

Comment:
i think this will be caught by 868, but I don't think there's any harm in repeating yourself 🤷

Reply:
For whatever reason the permissions config isn't getting picked up correctly in the tests...